waiter/src/waiter/state.clj

```diff
@@ -753,6 +754,54 @@
   (log/warn "[close-update-state-channel] closing update-state-chan of" service-id)
   (async/go (async/close! update-state-chan)))
 
+(defrecord InstanceTracker [service-id known-instance-ids scheduling-instance-timers starting-instance-trackers])
+
+(defn- new-tracker-val
+  [service-id]
+  (map->InstanceTracker {:service-id service-id
+                         :known-instance-ids #{}
+                         ; TODO - figure out min instances from service-id
+                         ; TODO - track overall request-to-running time
+                         :scheduling-instance-timers [[(timers/start (metrics/waiter-timer "app-schedule-time"))
+                                                       (timers/start (metrics/service-timer service-id "app-schedule-time"))]]
```

Comment:
Can we store the timers in a map instead of a vector? That way we can 'name' the timer contexts.

Reply:
Since we don't know the corresponding instance IDs at this point, I think a vector makes more sense than a map. I was considering using a queue, but queues in Clojure are very awkward.